src/Raven.Server/Documents/Sharding/ShardedDocumentDatabase.cs

```diff
@@ -226,6 +238,30 @@ public async Task DeleteBucketAsync(int bucket, long migrationIndex, string upto
         }
     }
 
+    private async Task WaitForConfirmationIndex(long confirmationIndex)
+    {
+        var cmd = new WaitForIndexNotificationCommand(new List<long> { confirmationIndex });
+        var tasks = new List<Task>(ShardingConfiguration.Orchestrator.Topology.Members.Count);
+        var clusterTopology = ServerStore.GetClusterTopology();
+
+        using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
```

Comment:
you are using the same context for all request _concurrently_. 
You need to create a context per request.

Reply:
fixed